### ISSUE TITLE
How to remove old Conbee stick?

### ISSUE THREAD

**Issue by TungstenE2** [dresden-elektronik/deconz-rest-plugin#2670]
Hi all,

I switched for Conbee I to Conbee II stick.
How to remove old Conbee from Deconz and Phoscon?

thx

![grafik](https://user-images.githubusercontent.com/39955685/78794824-81625200-79b4-11ea-929d-450fa584d446.png)
![grafik](https://user-images.githubusercontent.com/39955685/78795019-c38b9380-79b4-11ea-94e1-604d9fe7cf0e.png)

**Comment by Smanar** [comment]
I m interested too ^^.

How are you done the migration ? with backup/restore ?

**Comment by jensflorian** [comment]
I‘d look into the zll.db database file, make a backup copy and try to delete the records for Conbee I

**Comment by ebaauw** [comment]
That won’t do you any good, as this info doesn’t come from the database.  You’re seeing the deCONZ instances returned by the appspot server on discovery, before Phoscon has established an API connection to any instance.  The old entry should disappear in half an hour, unless its cached by your browser.

To delete the old coordinator node from the GUI, select it and press Del.

**Comment by TungstenE2** [comment]
Just deleting the coordinator did not do the trick. It reappears after restarting deCONZ. Any further ideas?

**Comment by ebaauw** [comment]
Might try and delete it from the `devices`table.

**Comment by TungstenE2** [comment]
how to delete from device table? any instructions on this?

**Comment by jensflorian** [comment]
See https://github.com/dresden-elektronik/deconz-rest-plugin/issues/833#issuecomment-427970554 - I recommend sqlitebrowser (https://sqlitebrowser.org). The database is located in 

> .local/share/dresden-elektronik/deCONZ

Always backup your database first, before manipulating it. You might also consider deleting the entries in `device_gui` table as well.

**Comment by TungstenE2** [comment]
so the issue is more than 2 years old and still not fixed?
old devices keep reapearing.

**Comment by TheWizz** [comment]
Yes, I see the same thing. While you can delete old CONBEE in DECONZ, it re-appears after next re-start. Even though you CAN probably delete it by hacking the database file, that should not be required since the UI does support deleting it (but doesn't apparently propagate that deletion to the database). So  this is a bug needs to be fixed.

-JM

**Comment by Mimiix** [comment]
Is this still a issue?

**Comment by MattL0** [comment]
YEs @manup

**Comment by MattL0** [comment]
I was not able to see it in the database.. Going there show my old conbee : https://phoscon.de/discover

[{"id":"00212EFFFF022CD5","internalipaddress":"192.168.0.100","macaddress":"xxxxxxxxxxxxxxxxx","internalport":xx,"name":"Conbee I","publicipaddress":"xxxxxxxxxxxx"}]

**Comment by MattL0** [comment]
How can this be resolved?

**Comment by MattL0** [comment]
> I was not able to see it in the database.. Going there shows my old conbee : https://phoscon.de/discover
> 
> [{"id":"00212EFFFF022CD5","internalipaddress":"192.168.0.100","macaddress":"xxxxxxxxxxxxxxxxx","internalport":xx,"name":"Conbee I","publicipaddress":"xxxxxxxxxxxx"}]
> 
> still there when I remove the database completly. I think this is bugged into the conbee 2 itself when I cloned it from conbee 1 last year

Any idea? @manup? I completly reformated my pc. Still the save

**Comment by Smanar** [comment]
I know you have reformated the PC, but have you make the try on another machine but without playing with datatase, on a clean install ?

When you make a backup there is too a file called sessions.sessions inside, you can take a look if you find the mac adress of the old conbee inside, if you don't find the mac adress on the database.

You haven't previous backup ?

**Comment by TungstenE2** [comment]
any news on this? No matter how often I delete old Conbee, it reappears after reboot.

Even updated to 2.08.00 raspbian-stretch-beta latest.

![grafik](https://user-images.githubusercontent.com/39955685/102801111-0fe8f900-43b5-11eb-838f-01d22014435f.png)

**Comment by Mimiix** [comment]
I've asked @manup to check it ^^:) Hope he replies soon!

**Comment by SwoopX** [comment]
This can currently only be done by deleting all traces in the database.

**Comment by TungstenE2** [comment]
so this is a bug in deconz for the delete option. Please have this fixed. Issue is open for 8 months and was even reported before. 

thx a lot

**Comment by Mimiix** [comment]
> so this is a bug in deconz for the delete option. Please have this fixed. Issue is open for 8 months and was even reported before.
> 
> thx a lot

I've created a internal issue on it with high priority for Manuel. 

Don't worry, I'll get him to reply.

**Comment by manup** [comment]
Currently the old coordinator node can only be deleted in the `zll.db` file and here the `devices` table, for example with SQLiteBrowser.

The delete node via GUI function is on the todo list to correctly delete everything related to a device.

Note having multiple coordinator notes doesn't cause harm other than looking odd.

Here is my coordinator graveyard:
![image](https://user-images.githubusercontent.com/383386/102824747-bcd76c00-43dd-11eb-8fdb-416f5ef644e7.png)

**Comment by TungstenE2** [comment]
thx for your feedback. It is good to hear that there is no harm, but from user perspective this is very iritating that a deleted node is reappearing after a reboot. So there is no much trust in the software, what else might be wrong.

eg I have a node which I renamed a couple of times, and the old name is reappearing after a reboot as well.
Is this a related issue, as the change in DB is not persitet?

![grafik](https://user-images.githubusercontent.com/39955685/102864697-dfe13a80-4434-11eb-93cd-26c232724809.png)

**Comment by TungstenE2** [comment]
Dear @manup thx for the fix in 2.9.0 for this issue.

May I ask how this was fixed?

In the change log you state 'Hide old inactive ccordinator nodes' rather than 'Remove old inactive ccordinator nodes'.

So am I right that you just do not dispaly inactive nodes anymore, but the notes still remain undeleted in the DB?
So the issue it self is not fixed, you just do not show it anymore?

![grafik](https://user-images.githubusercontent.com/39955685/103540523-205fa000-4e9a-11eb-8067-0746fbec6ae8.png)

**Comment by manup** [comment]
The change is related to just view the active coordinator at a time (since this is always true).

Note: Coordinators shouldn't be deleted in the database automatically, there are quite some setups out there, including mine, where different ConBee / RaspBee units are used over time. That's not an issue since the inactive entries do nothing.

The issue still open is the manual deletion of nodes and related resources via the context menu or Delete short-cut.
